extractor/text.go

```diff
@@ -1180,17 +1181,46 @@ var (
 // sortPosition sorts a text list by its elements' positions on a page.
 // Sorting is by orientation then top to bottom, left to right when page is orientated so that text
 // is horizontal.
+// Text is considered to be on different lines if the lines' orientedStart.Y differs by more than `tol`.
 func (pt *PageText) sortPosition(tol float64) {
+	if len(pt.marks) == 0 {
+		return
+	}
 	sort.SliceStable(pt.marks, func(i, j int) bool {
 		ti, tj := pt.marks[i], pt.marks[j]
 		if ti.orient != tj.orient {
 			return ti.orient < tj.orient
 		}
-		if math.Abs(ti.orientedStart.Y-tj.orientedStart.Y) > tol {
+		if ti.orientedStart.Y != tj.orientedStart.Y {
 			return ti.orientedStart.Y > tj.orientedStart.Y
 		}
 		return ti.orientedStart.X < tj.orientedStart.X
 	})
+
+	// Compute yOrder. yOrder increments if orientedStart.Y differs by more than `tol`.
+	yOrder := 0
+	pt.marks[0].yOrder = 0
+	y := pt.marks[0].orientedStart.Y
+	for i, t := range pt.marks[1:] {
+		if t.orientedStart.Y > y+tol {
+			yOrder = ((yOrder + 1000) / 1000) * 1000 // Next multiple of 1000
```

Comment:
Why in multiples of 1000 ?

Reply:
The multiple of 1000 was an arbitrary increment to give ordering ofg orientation then line number. I removed it to make the code more readable,